lib/posix/options/posix_clock.h

```diff
@@ -1,19 +1,82 @@
 /*
  * Copyright (c) 2023, Meta
+ * Copyright (c) 2025 Tenstorrent AI ULC
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #ifndef ZEPHYR_LIB_POSIX_POSIX_CLOCK_H_
 #define ZEPHYR_LIB_POSIX_POSIX_CLOCK_H_
 
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
 #include <time.h>
 
-#include <zephyr/kernel.h>
-#include <zephyr/posix/posix_types.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/sys/util.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static inline bool is_timespec_valid(const struct timespec *ts)
```

Comment:
Perhaps mark these with functions/macros below that are currently missing doxygen comments by `/** @cond INTERNAL_HIDDEN */`

Reply:
Fixed!